tests/forte/advanced_pipeline_test.py

```diff
@@ -280,5 +280,76 @@ def tearDown(self) -> None:
             os.remove(self._pl_config_path)
 
 
+class InferringOntologyTreeAutomaticallyTst(unittest.TestCase):
+    """
+    Test the Inferring Ontology Tree Automatically by using Recursive Method
+    """
+
+    def setUp(self) -> None:
+        self.expected = [
+            {
+                "entry_name": "ft.onto.wikipedia.WikiInfoBoxProperty",
+                "parent_entry": "forte.data.ontology.top.Generics",
+            },
+            {
+                "entry_name": "ft.onto.wikipedia.WikiInfoBoxMapped",
+                "parent_entry": "forte.data.ontology.top.Generics",
+            },
+            {
+                "entry_name": "ft.onto.wikipedia.WikiCategory",
+                "parent_entry": "forte.data.ontology.top.Generics",
+            },
+            {
+                "entry_name": "ft.onto.wikipedia.WikiPage",
+                "parent_entry": "forte.data.ontology.top.Annotation",
+            },
+            {
+                "entry_name": "ft.onto.wikipedia.WikiBody",
+                "parent_entry": "forte.data.ontology.top.Annotation",
+            },
+            {
+                "entry_name": "ft.onto.wikipedia.WikiSection",
+                "parent_entry": "forte.data.ontology.top.Annotation",
+            },
+            {
+                "entry_name": "ft.onto.wikipedia.WikiParagraph",
+                "parent_entry": "forte.data.ontology.top.Annotation",
+            },
+            {
+                "entry_name": "ft.onto.wikipedia.WikiTitle",
+                "parent_entry": "forte.data.ontology.top.Annotation",
+            },
+            {
+                "entry_name": "ft.onto.wikipedia.WikiArticleTitle",
+                "parent_entry": "forte.data.ontology.top.Annotation",
+            },
+            {
+                "entry_name": "ft.onto.wikipedia.WikiAnchor",
+                "parent_entry": "forte.data.ontology.top.Annotation",
+            },
+        ]
+
+        self.pipeline_1: Pipeline = Pipeline()
+        from ft.onto.wikipedia import WikiPage
+
+        self.pipeline_2: Pipeline = Pipeline()
+
+    def test_inferring_ontology_tree(self):
+        """
+        Check the correct of the results.
+        """
+        definitions_1 = self.pipeline_1.resource.get("onto_specs_dict")[
+            "definitions"
+        ]
+        definitions_2 = self.pipeline_2.resource.get("onto_specs_dict")[
+            "definitions"
+        ]
+        retDifference = [i for i in definitions_2 if i not in definitions_1]
+        retDifference_expected_union = [
+            i for i in self.expected if i in retDifference
+        ]
+        # self.assertEqual(len(self.expected), len(retDifference_expected_union))
```

Comment:
This test does not test anything -- If you want to skip a test, use @unittest.skip;